serving/src/test/java/feast/serving/it/ServingServiceIT.java

```diff
@@ -0,0 +1,555 @@
+  static List<List<byte[]>> featureReferenceWithTsByteArrays = new ArrayList<>();
+  static RedisProto.RedisKeyV2 redisKey = null;
+
+  static final int FEAST_SERVING_PORT = 6566;
+  @LocalServerPort private int metricsPort;
+
+  @ClassRule @Container
+  public static DockerComposeContainer environment =
+      new DockerComposeContainer(
+              new File("src/test/resources/docker-compose/docker-compose-it.yml"))
+          .withExposedService(
+              CORE,
+              FEAST_CORE_PORT,
+              Wait.forLogMessage(".*gRPC Server started.*\\n", 1)
+                  .withStartupTimeout(Duration.ofMinutes(SERVICE_START_MAX_WAIT_TIME_IN_MINUTES)))
+          .withExposedService(
+              JOB_CONTROLLER,
+              FEAST_JOB_CONTROLLER_PORT,
+              Wait.forLogMessage(".*gRPC Server started.*\\n", 1)
+                  .withStartupTimeout(Duration.ofMinutes(SERVICE_START_MAX_WAIT_TIME_IN_MINUTES)))
+          .withExposedService(REDIS, REDIS_PORT);
+
+  @BeforeAll
+  static void globalSetup() throws IOException, InitializationError, InterruptedException {
+    // Create Core client
+    coreClient = TestUtils.getApiClientForCore(FEAST_CORE_PORT);
+
+    // Create Redis client
+    RedisClient redisClient =
+        RedisClient.create(
+            new RedisURI(
+                environment.getServiceHost("redis_1", REDIS_PORT),
+                environment.getServicePort("redis_1", REDIS_PORT),
+                java.time.Duration.ofMillis(2000)));
+    StatefulRedisConnection<byte[], byte[]> connection = redisClient.connect(new ByteArrayCodec());
+    syncCommands = connection.sync();
+
+    String projectName = "default";
+    // Apply Entity
+    String entityName = "driver_id";
+    ValueProto.Value entityValue = ValueProto.Value.newBuilder().setInt64Val(1).build();
+    String description = "My driver id";
+    ValueProto.ValueType.Enum entityType = ValueProto.ValueType.Enum.INT64;
+    EntityProto.EntitySpecV2 entitySpec =
+        EntityProto.EntitySpecV2.newBuilder()
+            .setName(entityName)
+            .setDescription(description)
+            .setValueType(entityType)
+            .build();
+    TestUtils.applyEntity(coreClient, projectName, entitySpec);
+
+    // Apply FeatureTable
+    String featureTableName = "rides";
+    List<String> entities =
+        new ArrayList<>() {
+          {
+            add(entityName);
+          }
+        };
+    HashMap<String, ValueProto.ValueType.Enum> features = new HashMap<>();
+
+    // Feature 1
+    String feature1 = "trip_cost";
+    ValueProto.Value feature1Value = ValueProto.Value.newBuilder().setDoubleVal(42.2).build();
+    ServingAPIProto.FeatureReferenceV2 feature1Reference =
+        ServingAPIProto.FeatureReferenceV2.newBuilder()
+            .setFeatureTable(featureTableName)
+            .setName(feature1)
+            .build();
+
+    // Feature 2
+    String feature2 = "trip_distance";
```

Comment:
a lot of repeating code. Please move this to / reuse existing from `common-test`